workspaces/global-header/plugins/global-header/src/components/SearchComponent/SearchBar.tsx

```diff
@@ -0,0 +1,151 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {
+  SearchResultState,
+  SearchResultProps,
+} from '@backstage/plugin-search-react';
+import Typography from '@mui/material/Typography';
+import { Link } from '@backstage/core-components';
+import ListItem from '@mui/material/ListItem';
+import Box from '@mui/material/Box';
+import Divider from '@mui/material/Divider';
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
+import Autocomplete from '@mui/material/Autocomplete';
+import TextField from '@mui/material/TextField';
+import InputAdornment from '@mui/material/InputAdornment';
+import SearchIcon from '@mui/icons-material/Search';
+import { highlightMatch } from '../../utils/stringUtils';
+import styles from './SearchBar.module.css';
+import { useNavigate } from 'react-router-dom';
+
+interface SearchBarProps {
+  query: SearchResultProps['query'];
+  setSearchTerm: (term: string) => void;
+}
+export const SearchBar = (props: SearchBarProps) => {
+  const { query, setSearchTerm } = props;
+  const navigate = useNavigate();
+
+  return (
+    <SearchResultState {...props}>
+      {({ loading, error, value }) => {
+        const results = query?.term ? value?.results ?? [] : [];
+        let options: string[] = [];
+        if (results.length > 0) {
+          options = [
+            ...results.slice(0, 5).map(result => result.document.title),
+            `${query?.term}`,
+          ];
+        } else if (query?.term) {
+          options = ['No results found'];
+        }
+
+        return (
+          <Autocomplete
+            freeSolo
+            options={options}
+            loading={loading}
+            getOptionLabel={option => option ?? ''}
+            onInputChange={(_, inputValue) => setSearchTerm(inputValue)}
+            sx={{ width: '100%' }}
+            filterOptions={x => x}
+            getOptionDisabled={option => option === 'No results found'}
+            onKeyDown={event => {
+              if (event.key === 'Enter') {
+                event.preventDefault();
+                if (query?.term) {
+                  navigate(`/search?query=${query?.term}`);
+                }
+              }
+            }}
+            renderInput={params => (
+              <TextField
+                {...params}
+                placeholder="Search..."
+                variant="standard"
+                error={!!error}
+                helperText={error ? 'Error fetching results' : ''}
+                InputProps={{
+                  ...params.InputProps,
+                  disableUnderline: true,
+                  startAdornment: (
+                    <InputAdornment position="start">
+                      <SearchIcon style={{ color: '#fff' }} />
+                    </InputAdornment>
+                  ),
+                }}
+                sx={{
+                  pt: '0.5rem',
```

Comment:
For me, it feels like this padding-top breaks that the input field is vertically centered.

```suggestion
```

Reply:
To be honest, for me, the first one looks the best. :smirk: